grudge/op.py

```diff
@@ -403,155 +1002,153 @@ def norm(dcoll, vec, p, dd=None):
         else:
             raise ValueError("unsupported norm order")
 
-    return _norm(dcoll, p, dd)(arg=vec)
-
-
-@memoize_on_first_arg
-def _nodal_reduction(dcoll, operator, dd):
-    return bind(dcoll, operator(dd)(sym.var("arg")), local_only=True)
+    return _norm(dcoll, vec, p, dd)
 
 
 def nodal_sum(dcoll, dd, vec):
-    return _nodal_reduction(dcoll, sym.NodalSum, dd)(arg=vec)
+    from warnings import warn
+    warn("Using 'nodal_sum' is deprecated, use 'nodal_summation' instead.",
```

Comment:
The only reason I did so is because `dd` is not needed. But I'm rethinking this now and I think it's better to just leave them as is. I'll update this

Reply:
https://github.com/inducer/grudge/pull/74/commits/d9bb98d696a00f727a4c29166f0debdfd1019b96